Backend/ServiceLayer/Services/ExceptionService.cs

```diff
@@ -12,6 +12,7 @@ public class InvalidEmailException : Exception
         {
             public InvalidEmailException() { }
             public InvalidEmailException(string message) : base(message) { }
+            public InvalidEmailException(string message, Exception innerException) : base(message, innerException) { }

```

Comment:
Updated to include an inner exception so the stack trace is not lost on creation after a catch.